src/main/java/world/bentobox/parkour/listeners/CourseRunnerListener.java

```diff
@@ -71,11 +72,40 @@ public void onVisitorArrive(IslandEnterEvent e) {
         } else if (!parkourRunManager.timers().containsKey(e.getPlayerUUID())) {
             user.notify("parkour.to-start");
         }
-        if (island.getFlag(addon.PARKOUR_CREATIVE) <= island.getRank(user)) {
-            user.setGameMode(GameMode.CREATIVE);
-        } else {
-            user.setGameMode(GameMode.SURVIVAL);
+
+        if (addon.inWorld(user.getLocation())) {
+            // only affect gamemode if they are currently in the parkour world
+            // cross world teleportation is handled in onPlayerChangeWorld
+            if (island.getFlag(addon.PARKOUR_CREATIVE) <= island.getRank(user)) {
+                user.setGameMode(GameMode.CREATIVE);
+            } else {
+                user.setGameMode(GameMode.SURVIVAL);
+            }
         }
+
+
+    }
+
+    @EventHandler
+    public void onPlayerChangeWorld(PlayerChangedWorldEvent event) {
+        User user = User.getInstance(event.getPlayer());
+        Location currentLocation = user.getLocation();
+        if (!addon.inWorld(currentLocation)) {
```

Comment:
This includes nether and end.

Reply:
If a player changes world into any parkour managed world, it should still change gamenode, right?

Reply:
Yes. I'm just thinking if you need to consider the use case where someone is in the Parkour Nether, but teleports back to the Parkour Overworld. I anticipate Parkour courses that cross dimensions. That's why.

Reply:
At the moment, cross dimensional travel is blocked by the teleport event iirc, is that also a bug?

Reply:
stopped portal teleports from stopping run, added a check that stops changing the gamemode if you are coming from a parkour world